exo-setup/src/docker-setup.ls

```diff
@@ -1,67 +1,82 @@
-    @_build-docker-image done
-
-
-  _build-docker-image: (done) ~>
-    new ObservableProcess(call-args(DockerHelper.get-build-command author: @service-config.author, name: dashify(@service-config.type))
-                          cwd: @config.root
-                          stdout: {@write}
-                          stderr: {@write})
-      ..on 'ended', (exit-code, killed) ~>
-        | exit-code is 0  =>  @logger.log {@role, text: "Docker setup finished"}
-        | otherwise       =>
-          @logger.log {@role, text: "Docker setup failed"}
-          process.exit exit-code
-        done!
-
-
-  _docker-file-exists: ~>
-    try
-      fs.exists-sync path.join(@config.root, 'Dockerfile')
-    catch
-      no
-
-
-  _setup-external-service: (done) ~>
-    throw new Error red "No location or docker-image specified" unless @config.docker-image
-    image = @config.docker-image |> (.split '/')
-    new ObservableProcess((DockerHelper.get-pull-command author: image[0], name: image[1]),
-                          stdout: {@write}
-                          stderr: {@write})
-      ..on 'ended', (exit-code) ~>
-        | exit-code isnt 0  =>
-          @logger.log {@role, text: 'Docker setup failed'}
-          process.exit exit-code
-        | _                 =>  done!
-
-
-  write: (text) ~>
-    @logger.log {@role, text, trim: yes}
-
-
-
+# Renders docker-compose.yml file with service configuration
+class DockerSetup
+
+  ({@app-name, @role, @logger, @service-location, @docker-image}) ->
+    @service-config = yaml.safe-load fs.read-file-sync(path.join(process.cwd!, @service-location, 'service.yml'), 'utf8') if @service-location
+
+
+  get-service-docker-config: ~> 
+    | @service-config => @_setup-service!
+    | otherwise       => @_setup-external-service!
+
+
+  _setup-service: ->
+    docker-config = {}
+    docker-config[@role] =
+      Obj.compact do
+        build: @service-location 
+        command: @service-config.startup.command
+        ports: @service-config.docker?.ports or null
+        links: @_get-service-links!
+        environment: @_get-environment-vars! 
+        depends_on: @_get-service-dependencies!
+    for dependency, dependency-config of @service-config.dependencies
+      docker-config[dependency + dependency-config.dev.version] = @_setup-service-dependencies dependency, dependency-config.dev
+    docker-config
+
+
+  _get-service-links: ->
+    links = []
+    for dependency, dependency-config of @service-config.dependencies
+      links.push "#{dependency + dependency-config.dev.version}:#{dependency}"
+    if links.length then links else null
+
+
+  _get-environment-vars: ->
+    env-vars =
+      ROLE: @role
+      EXOCOM_HOST: 'exocom'
+      EXOCOM_PORT: '$EXOCOM_PORT'
+    for dependency of @service-config.dependencies
+      env-vars[dependency.to-upper-case!] = dependency
+    env-vars
+
+
+  _get-service-dependencies: ->
+    dependencies = ['exocom']
```

Comment:
Is it possible that we have some services that aren't at all connected to `exocom` (or any other message bus) in the future? Not that it would need to be addressed right now, just something to consider.

@kevgo thoughts?